kratos/factories/factory.h

```diff
@@ -0,0 +1,251 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+#if !defined(KRATOS_FACTORY_H_INCLUDED )
+#define  KRATOS_FACTORY_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "includes/kratos_components.h"
+#include "includes/shared_pointers.h"
+
+namespace Kratos
+{
+///@name Type Definitions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class FactoryMethods
+ * @ingroup KratosCore
+ * @brief Here we define some common methods
+ * @details Defines the base class factory methods
+ * @author Vicente Mataix Ferrandiz
+ */
+class FactoryMethods
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of FactoryMethods
+    KRATOS_CLASS_POINTER_DEFINITION(FactoryMethods);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /** Constructor.
+     */
+    explicit FactoryMethods(){}
```

Comment:
Why is not FactoryBase as it only defines the interface and work as base type without the type template?

Reply:
Huuum, let me check

Reply:
Because the Has method, requires to know the class, I can move the template to the has method and unify the classes, as you wish

Reply:
I have been thinking, and if I remove the template the exposition to python will be broken. I think it is better to keep it as it is

Reply:
sol let's rename it to FactoryBase. No?

Reply:
Done